app/server/src/features/accounts/methods.unit.test.ts

```diff
@@ -0,0 +1,243 @@
+import bcrypt from 'bcryptjs';
+import * as AccountMethods from './methods';
+import { prismaMock } from '../../../mocks/prisma/singleton';
+import * as jwt from '@local/lib/jwt';
+import { toGlobalId } from '../utils';
+import { faker } from '@faker-js/faker';
+
+const toUserId = toGlobalId('User');
+const userTextPassword = faker.internet.password();
+const userData = {
+    id: faker.datatype.uuid(),
+    createdAt: new Date(),
+    email: faker.internet.email(),
+    firstName: faker.name.firstName(),
+    lastName: faker.name.lastName(),
+    fullName: '',
+    password: userTextPassword,
+    preferredLang: 'EN',
+    canMakeOrgs: false,
+};
+
+beforeAll(async () => {
+    userData.password = await bcrypt.hash(userData.password, 10);
+    userData.fullName = userData.firstName + ' ' + userData.lastName;
+});
+
+/*
+ *  test cases for registering a user optionally with a password
+ */
+describe('register', () => {
+    test('should register new user with a password', async () => {
+        const expectedOutput = {
+            ...userData,
+        };
+
+        prismaMock.user.create.mockResolvedValueOnce(expectedOutput);
+
+        const output = await AccountMethods.register(prismaMock, userData, userTextPassword);
+
+        expect(output).toEqual(expectedOutput);
+    });
+
+    test('should register new user with null password', async () => {
+        const emptyPassword = '';
+
+        const expectedOutput = {
+            ...userData,
+            password: await bcrypt.hash(emptyPassword, 10),
+        };
```

Comment:
Is this passing?  It look like if we pass an empty password, that it's null, so why is there a `bcrypt.hash` in the `expectedOutput` & then why is that passing if it is? 

https://github.com/c2c-project/prytaneum/blob/519521f642d93a47a2eeaacf14ff20efe6bc7fe8/app/server/src/features/accounts/methods.ts#L17

Reply:
yes all the tests are passing, I think it's mainly the name that's an issue, I'll change the test name to `should register new user with empty string password` because it's testing with an empty string not null. bcrypt will hash an empty string but not a null value.

Reply:
I'm not sure I fully understand.  From my understanding, this test doesn't make sense for a few reasons:
1. We never have an empty password in the database.
2. We never pass in an empty string as the password.  
However, both of these things are done in the test.

Reply:
I think that was done because we originally were going to make accounts with blank passwords for mass invites and Kevin wanted us to create the account without a password that could be set up later, but we could probably just make the accounts with a default password that is emailed to them instead for better security.

Reply:
Yes, you are correct that we make accounts without a password for accounts that we generate based off of an invite list.  However, it's an important distinction between an account with a null password versus one with an empty string.  If you look at the test, it is encrypting an empty password and using that as the expected data.  That should NOT be passing because that is not a valid value.  The only valid values are either a) a null password or b) an encrypted password string.

Reply:
Oh I see now, I updated it to actually check the null.